crates/uv-resolver/src/resolver/locals.rs

```diff
@@ -0,0 +1,165 @@
+                // _know_ `foo @ 1.0.0+local` is required and would conflict.
+                if is_compatible(local, specifier.version()) {
+                    VersionSpecifier::new(Operator::LessThanEqual, local.clone())
+                } else {
+                    specifier.clone()
+                }
+            }
+            Operator::GreaterThan => {
+                // Given `foo>1.0.0`, if the local version is `1.0.0+local`, map to
+                // `foo>1.0.0+local`. This has the intended effect of disallowing `1.0.0+local`.
+                if is_compatible(local, specifier.version()) {
+                    VersionSpecifier::new(Operator::GreaterThan, local.clone())
+                } else {
+                    specifier.clone()
+                }
+            }
+            Operator::ExactEqual => {
+                // Given `foo===1.0.0`, `1.0.0+local` is already disallowed.
+                specifier.clone()
+            }
+            Operator::TildeEqual => {
+                // Given `foo~=1.0.0`, `foo~=1.0.0+local` is already allowed.
+                specifier.clone()
+            }
+            Operator::LessThan => {
+                // Given `foo<1.0.0`, `1.0.0+local` is already disallowed.
+                specifier.clone()
+            }
+            Operator::GreaterThanEqual => {
+                // Given `foo>=1.0.0`, `foo>1.0.0+local` is already allowed.
+                specifier.clone()
+            }
+        }
+    }
+}
+
+/// Returns `true` if a provided version is compatible with the expected local version.
+///
+/// The versions are compatible if they are the same including their local segment, or the
+/// same except for the local segment, which is empty in the provided version.
+///
+/// For example, if the expected version is `1.0.0+local` and the provided version is `1.0.0+other`,
+/// this function will return `false`.
+///
+/// If the expected version is `1.0.0+local` and the provided version is `1.0.0`, the function will
+/// return `true`.
+fn is_compatible(expected: &Version, provided: &Version) -> bool {
+    // The requirements should be the same, ignoring local segments.
+    if expected.clone().without_local() != provided.clone().without_local() {
+        return false;
+    }
+
+    // If the provided version has a local segment, it should be the same as the expected
+    // version.
+    if provided.local().is_empty() {
+        true
+    } else {
+        expected.local() == provided.local()
+    }
+}
+
+/// If a [`VersionSpecifier`] represents exact equality against a local version, return the local
+/// version.
+fn to_local(specifier: &VersionSpecifier) -> Option<&Version> {
+    if !matches!(specifier.operator(), Operator::Equal | Operator::ExactEqual) {
+        return None;
+    };
+
+    if specifier.version().local().is_empty() {
+        return None;
+    }
+
+    Some(specifier.version())
+}
```

Comment:
What do you think about adding some units tests for the routines in this module?

Reply:
Good call.